src/qt/diagnosticsdialog.cpp

```diff
@@ -89,7 +89,7 @@ int DiagnosticsDialog::VerifyCPIDIsInNeuralNetwork() {
     if(beacons.length() < 100)
         return -1;
 
-    std::string cpid = GetArgument("cpid", "");
+    std::string cpid = msPrimaryCPID;
     if(cpid != "" && beacons.find(cpid) != std::string::npos)
```

Comment:
Change this to something like:

```
int DiagnosticsDialog::VerifyCPIDIsInNeuralNetwork()
{
    std::string beacons = GetListOf("beacon");
    boost::algorithm::to_lower(beacons);
    return IsResearcher(cpid) && beacons.find(cpid) != std::string::npos;
}
```

Maybe we should change all return values to bool unless we need to track multiple return codes.

Reply:
yea i can go boolerific